azurerm/internal/services/apimanagement/tests/api_management_api_resource_test.go

```diff
@@ -575,31 +572,82 @@ resource "azurerm_api_management_api" "test" {
 `, template, data.RandomInteger, data.RandomInteger)
 }
 
-func testAccAzureRMApiManagementApi_template(data acceptance.TestData) string {
+func testAccAzureRMApiManagementApi_oauth2Authorization(data acceptance.TestData) string {
+	template := testAccAzureRMApiManagementApi_template(data)
 	return fmt.Sprintf(`
-provider "azurerm" {
-  features {}
+%s
+
+resource "azurerm_api_management_authorization_server" "test" {
+  name                         = "acctestauthsrv-%d"
+  resource_group_name          = azurerm_resource_group.test.name
+  api_management_name          = azurerm_api_management.test.name
+  display_name                 = "Test Group"
+  authorization_endpoint       = "https://azacctest.hashicorptest.com/client/authorize"
+  client_id                    = "42424242-4242-4242-4242-424242424242"
+  client_registration_endpoint = "https://azacctest.hashicorptest.com/client/register"
+
+  grant_types = [
+    "implicit",
+  ]
+
+  authorization_methods = [
+    "GET",
+  ]
 }
 
-resource "azurerm_resource_group" "test" {
-  name     = "acctestRG-%d"
-  location = "%s"
+resource "azurerm_api_management_api" "test" {
+  name                = "acctestapi-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  api_management_name = azurerm_api_management.test.name
+  display_name        = "api1"
+  path                = "api1"
+  protocols           = ["https"]
+  revision            = "1"
+  version             = "v1"
```

Comment:
This line is causing the service validation to fail (internal validation that needs `version_set_id` to be set.  It's optional, so I'd suggest:
```suggestion
```